bulk-import-tools/internal/orchcli/orchcli.go

```diff
@@ -243,6 +246,43 @@ func (oC *OrchCli) InstanceExists(ctx context.Context, sn, uuid string) (bool, e
 	return false, nil
 }
 
+func (oC *OrchCli) GetHostID(ctx context.Context, sn, uuid string) (string, error) {
+	if sn == "" && uuid == "" {
+		return "", e.NewCustomError(e.ErrInternal)
+	}
+
+	uParsed := *oC.SvcURL
+	uParsed.Path = path.Join(uParsed.Path, fmt.Sprintf("/v1/projects/%s/compute/hosts", oC.Project))
+	query := uParsed.Query()
+	query.Set("filter", fmt.Sprintf("%s=%q AND %s=%q", "serialNumber", sn, "uuid", uuid))
```

Comment:
shall we do 2 separate queries and merge the results and potentially spot inconsistencies ?

Reply:
Or build the query properly, based on what is being requested, SN and UUID, SN only, UUID only

Reply:
Do you intend to say that the BIT should be able to recover entries that are incomplete? For example - 
1.) A user manually creates an entry from web-ui providing only Serial No. ABCD1234
2.) Same/different user tries to bulk import & also includes the Serial No ABCD1234 but also provides a corresponding UUID

Option 1- Another host added with both the fields. SN is not unique
Option 2- No action on row, Only duplicate reported 
Option 3- Details merged into a single host

Current implementation is doing Option 1. The idea being that our backend allows hosts to have duplicate Serial No & hence it might be a genuine case of duplicate Serial No & the user intends to create 2 entries.

Reply:
my first concern is that you are doing a blind AND and potentially not finding the host. I am also ok to implement what Daniele suggested if you have ontext to build the right query. Otherwise do a query by UUID and do a query by SN and hopefully you will get the host you are looking for or multiple hosts and warn the user.

Reply:
![Screenshot 2025-05-06 193805](https://github.com/user-attachments/assets/38b32fe1-999f-442a-a964-ebbd7277473c)
Greens symbolic of expected outcome.